zqd/storage/archivestore/archivestore.go

```diff
@@ -113,10 +113,26 @@ func (s *Storage) Summary(_ context.Context) (storage.Summary, error) {
 	return sum, nil
 }
 
+func (s *Storage) Write(ctx context.Context, zctx *resolver.Context, zr zbuf.Reader) error {
+	_, err := s.ark.UpdateCheck()
+	if err != nil {
+		return err
+	}
+	return archive.Import(ctx, s.ark, zctx, zr)
+}
+
 func (s *Storage) IndexSearch(ctx context.Context, zctx *resolver.Context, query archive.IndexQuery) (zbuf.ReadCloser, error) {
+	_, err := s.ark.UpdateCheck()
+	if err != nil {
+		return nil, err
+	}
 	return archive.FindReadCloser(ctx, zctx, s.ark, query, archive.AddPath(archive.DefaultAddPathField, false))
 }
 
 func (s *Storage) ArchiveStat(ctx context.Context, zctx *resolver.Context) (zbuf.ReadCloser, error) {
+	_, err := s.ark.UpdateCheck()
```

Comment:
(similar to above, the SpanWalk inside archive.Stat should do this already)

Reply:
Doesn't look like it is. Tests on ArchiveStat and ArchiveIndexSearch fail when I remove these. I'll see about adding something like this in SpanWalk.

Reply:
Huh, it is being called. Digging...